packages/snap-preact-components/src/components/Templates/Search/Search.test.tsx

```diff
@@ -0,0 +1,318 @@
+
+		expect(element).toBeInTheDocument();
+		expect(button).toBeInTheDocument();
+		expect(sidebar).toBeInTheDocument();
+		expect(button).toHaveTextContent(buttonText);
+
+		expect(sidebar?.parentElement?.parentElement).toHaveClass('ss__dropdown--open');
+		await userEvent.click(button!);
+		await waitFor(() => {
+			expect(sidebar?.parentElement?.parentElement).not.toHaveClass('ss__dropdown--open');
+		});
+		await userEvent.click(button!);
+		await waitFor(() => {
+			expect(sidebar?.parentElement?.parentElement).toHaveClass('ss__dropdown--open');
+		});
+	});
+
+	it('shows mobilesider instead of sidebar on mobile', () => {
+		//override matchmedia to always return true
+		Object.defineProperty(window, 'matchMedia', {
+			writable: true,
+			value: jest.fn().mockImplementation((query) => ({
+				matches: true, // always set as true
+				media: query,
+				onchange: null,
+				addListener: jest.fn(), // Deprecated
+				removeListener: jest.fn(), // Deprecated
+				addEventListener: jest.fn(),
+				removeEventListener: jest.fn(),
+				dispatchEvent: jest.fn(),
+			})),
+		});
+
+		const rendered = render(<Search controller={controller} />);
+		const element = rendered.container.querySelector('.ss__search')!;
+		const sidebar = rendered.container.querySelector('.ss__sidebar');
+		const mobileSidebar = rendered.container.querySelector('.ss__mobile-sidebar__slideout');
+
+		expect(element).toBeInTheDocument();
+		expect(mobileSidebar).toBeInTheDocument();
+		expect(sidebar).not.toBeInTheDocument();
+	});
+
+	it('renders with merchandising banners', async () => {
+		mockClient.mockData.updateConfig({ search: 'merchandising' });
+
+		await controller.search();
+
+		const rendered = render(<Search controller={controller} />);
+
+		const element = rendered.container.querySelector('.ss__search');
+		const banners = rendered.container.querySelectorAll('.ss__banner');
+		const headerBanner = rendered.container.querySelector('.ss__banner--header');
+		const bannerBanner = rendered.container.querySelector('.ss__banner--banner');
+		const footerBanner = rendered.container.querySelector('.ss__banner--footer');
+
+		expect(element).toBeInTheDocument();
+		expect(banners).toHaveLength(3);
+		expect(headerBanner).toBeInTheDocument();
+		expect(bannerBanner).toBeInTheDocument();
+		expect(footerBanner).toBeInTheDocument();
+	});
+
+	it('can hide all merchandising banners', async () => {
+		await controller.search();
+
+		const rendered = render(<Search controller={controller} hideMerchandisingBanners />);
+
+		const element = rendered.container.querySelector('.ss__search');
+		const banners = rendered.container.querySelectorAll('.ss__banner');
+
+		expect(element).toBeInTheDocument();
+		expect(banners).toHaveLength(0);
+	});
+
+	it('can hide only some merchandising banners', async () => {
+		await controller.search();
+
+		const rendered = render(<Search controller={controller} hideMerchandisingBanners={[ContentType.FOOTER, ContentType.HEADER]} />);
```

Comment:
Do we have to use enums here? Just curious how this would work in the integration.